include/arch/gdbstub.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2020 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_ARCH_GDBSTUB_H_
+#define ZEPHYR_INCLUDE_ARCH_GDBSTUB_H_
+
+#include <inttypes.h>
+
+#if defined(CONFIG_X86)
+#include <arch/x86/ia32/gdbstub.h>
+#endif
+
+/**
+ * ARCH_GDB_NUM_REGISTERS is architecure specific and
+ * this symbol must be defined in architecure specific header
+ */
+#define GDB_NUM_REGISTERS ARCH_GDB_NUM_REGISTERS
```

Comment:
please see how other ARCH_ macros are documented in include/sys/arch_interface.h